File: go/scrabble-score/scrabble_score.go
package scrabble

import "unicode"

var letterScoresCoarse = map[string]int{
	"AEIOULNRST": 1,
	"DG":         2,
	"BCMP":       3,
	"FHVWY":      4,
	"K":          5,
	"JX":         8,
	"QZ":         10,
}

func Score(word string) int {
	letterScoresGranular := make(map[rune]int)
	for key, val := range letterScoresCoarse {
		for _, ch := range key {
			letterScoresGranular[ch] = val
		}
	}
	result := 0
	for _, ch := range word {
		upperCh := unicode.ToUpper(ch)
		result += letterScoresGranular[upperCh]
	}
	return result
}

File: cpp/scrabble-score/scrabble_score.cpp
#include "scrabble_score.h"
#include <cctype>

namespace scrabble_score {

    int score(const std::string &word) {
        int score = 0;
        for (const auto &ch : word) {
            score += SCRABBLE_VALUES[std::tolower(ch) - 'a'];
        }
        return score;
    }

}  // namespace scrabble_score
